pkg/didcomm/protocol/route/service.go

```diff
@@ -427,11 +441,24 @@ func (s *Service) AddKey(recKey string) error {
 	}
 
 	// remove the channel once its been processed
-	s.setRouteRegistrationCh(msgID, nil)
+	s.setKeyUpdateResponseCh(msgID, nil)
 
 	return nil
 }
 
+// Config fetches the router config - endpoint and routingKeys.
+func (s *Service) Config() (*Config, error) {
+	// check if router is already registered
+	_, err := s.getRouterConnectionID()
```

Comment:
I assume this case cannot be determined by the call to s.getRouterConfig()?

Reply:
well, we can check if route configs are present or not. here, i feel better to check the source of truth for validation, since all the subsequent actions like AddKey, RemoveKey are dependent on this data.